target/linux/ramips/dts/mt7628an_joowin_jw-wr758ac.dtsi

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "mt7628an.dtsi"
+
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	compatible = "joowin,jw-wr758ac", "mediatek,mt7628an-soc";
+
+	keys {
+		compatible = "gpio-keys";
+
+		wps {
+			label = "wps";
+			gpios = <&gpio 38 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		wifi-high {
+			label = "blue:wifi-high";
+			gpios = <&gpio 11 GPIO_ACTIVE_LOW>;
+		};
+
+		wifi-med {
+			label = "blue:wifi-med";
+			gpios = <&gpio 44 GPIO_ACTIVE_LOW>;
+		};
+
+		wifi-low {
+			label = "blue:wifi-low";
+			gpios = <&gpio 37 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&state_default {
+	gpio {
+		groups = "i2c", "wled_an", "gpio", "refclk", "wdt", "p0led_an";
+		function = "gpio";
+	};
+};
+
+&pcie {
+	status = "okay";
+};
+
+&pcie0 {
+	wifi@0,0 {
+		compatible = "mediatek,mt76";
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x8000>;
+		ieee80211-freq-limit = <5000000 6000000>;
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <50000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x30000>;
+				read-only;
+			};
+
+			partition@30000 {
+				label = "u-boot-env";
+				reg = <0x30000 0x10000>;
+				read-only;
+			};
+
+			factory: partition@40000 {
+				label = "factory";
+				reg = <0x40000 0x10000>;
+				read-only;
+			};
+
+			partition@50000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x50000 0x760000>;
+			};
+		};
+	};
+};
+
+&wmac {
+	status = "okay";
+};
+
+&ethernet {
+	nvmem-cells = <&macaddr_factory_e000>;
+	nvmem-cell-names = "mac-address";
+};
+
+&esw {
+	mediatek,portmap = <0x2f>;
```

Comment:
Do you need this line? This line means that port 4 is for wan, contrary to what you described in `02_network`.

Reply:
Hi. Sorry for the late reply.

I wasn't sure what to put there, this was based on another dts I tried to make in the past for another device that had a different number of ports.

The joowin WR758AC devices only have one port, so the `02_network` file has the correct information. Should I just remove the `&esw` section then?

I've noticed meanwhile that there are already some other devices with the same switch definition:
```
        tplink,tl-wr902ac-v3|\
        wavlink,wl-wn576a2)
                ucidef_add_switch "switch0" \
                        "4:lan" "6@eth0"
                ;;
```
Should I move the joowin definitions close to those? Or, for alphabetical purposes, should I keep them as they are?

Thanks in advance.

Reply:
Meanwhile, for example, the mt7628an_wavlink_wl-wn577a2 device also has `mediatek,portmap = <0x2f>;` so it must be also wrong?
Thanks in advice.

Reply:
> The joowin WR758AC devices only have one port, so the `02_network` file has the correct information. Should I just remove the `&esw` section then?

My suggestion:
```
&esw {
	mediatek,portdisable = <0x2f>;
};
```
`mediatek,portdisable` is for turning off unused ports.

> Should I move the joowin definitions close to those? Or, for alphabetical purposes, should I keep them as they are?

It should be like this:
```diff
+	joowin,jw-wr758ac-v1|\
+	joowin,jw-wr758ac-v2|\
	tplink,tl-wr902ac-v3|\
	wavlink,wl-wn576a2)
		ucidef_add_switch "switch0" \
			"4:lan" "6@eth0"
		;;
```

> Meanwhile, for example, the mt7628an_wavlink_wl-wn577a2 device also has `mediatek,portmap = <0x2f>;` so it must be also wrong? Thanks in advice.

Probably.

Reply:
Fixed in latest push.